sumatra/core.py

```diff
@@ -105,8 +125,67 @@ def register(self, component):
                     name = component.name
                 else:
                     name = component.__name__
-                self.components[base_class][name] = component
+                self._components[base_class][name] = component
                 return
         raise TypeError("%s is not a Sumatra component." % component)
 
-registry = Registry()
+
+def component_type(cls):
```

Comment:
I wonder whether that name should be distinguished more from ``component`` or have a leading underscore so to not be confused by someone writing a plug-in.